Source: Ali Polatel
Upstream: http://bugs.python.org/issue1254718
Reason: Fix build/link faults when using ccache


From 6ac197f38d05a7e81c6337d567e8eed4f13a4d5b Mon Sep 17 00:00:00 2001
From: Ali Polatel <hawking@gentoo.org>
Date: Sun, 26 Oct 2008 16:20:21 +0200
Subject: [PATCH 08/10] Fix distutils's detection of gcc.

---
 Lib/distutils/unixccompiler.py |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index bccb952..1f9f1de 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -294,7 +294,8 @@ class UnixCCompiler(CCompiler):
             return "+s -L" + dir
         elif sys.platform[:7] == "irix646" or sys.platform[:6] == "osf1V5":
             return ["-rpath", dir]
-        elif compiler[:3] == "gcc" or compiler[:3] == "g++":
+        # This should recognize gcc, g++, gcc-x.y, ccache gcc, etc.
+        elif compiler.find("gcc") >= 0 or compiler.find("g++") >= 0:
             return "-Wl,-R" + dir
         else:
             return "-R" + dir
-- 
1.6.0.2

